### DISCUSSION TITLE
Jackett implementation, is it working?

### DISCUSSION THREAD

**Discussion by Mahava** [LivingWithHippos/unchained-android#358] [Feedback]
Hi LWH,

I set up the jackett creds. in the dev branch but the api token doesnt stick.   And how would it be used to search anyway?  Is there going to be a plugin for Jackett?

**Comment by Invictaz** [comment]
@Mahava I have the dev build but cannot see where you have to enter Jackett creds? Where do you see those?

**Reply by LivingWithHippos** [reply]
There is no way to search on jackett yet for now, I'm still implementing it, it's probably in dev because I merged some branches with a partial work on it

**Comment by LivingWithHippos** [comment]
I'm going to put two tabs on top of the search page like for the torrent/downloads page. One is going to use the search plugins, the other will allow you to search on a torznab compatible server such as jackett. You will need to self host one since they are used privately only from what I know.
Right now I can't remember the status on dev, but the branch I'm working on is https://github.com/LivingWithHippos/unchained-android/tree/feat/jackett which still needs some work. I tried with a jackett server and it works so now I need to create all the logic and ui, while also reworking the plugins tab.

The todo is:
- [ ] rework the plugin page adding two tabs on top
- [ ] rework the plugin tab, adding a sidesheet with all the settings to unclutter a bit the search screen (in progress)
- [ ] create the "service" tab which will allow you to search on torznab
- [ ] connect everything

I'll also try to add some screenshots or a figma mockup so I can get feedback on the ui before implementing everything

**Comment by Invictaz** [comment]
@LivingWithHippos Can we host it online? I'm not familiar with Docker or anything but do we need a VPS or Raspberry Pi to host stuff? Or can we use a server on our phone in Termux?

**Reply by LivingWithHippos** [reply]
It can be anywhere as long as you can reach it from your phone, so if you can make it run on your phone with termux you can probably use 127.0.0.1 as ip and it should work. Otherwise you can use a pc/raspberry in your home so you don't need to expose it to the internet (but then you'll be able to use it only on the same network at home). VPS is also an option, you'll need to be careful exposing it to the internet

**Reply by Invictaz** [reply]
It needs to be exposed to the internet in order to reach the API from various torrent indexers.

**Reply by LivingWithHippos** [reply]
Yes I meant that you cannot reach jackett from the outside if you host it on your phone or your home network so it's safer
